plugins/modules/web_infrastructure/bugzilla.py

```diff
@@ -0,0 +1,227 @@
+
+  bug:
+    required: false
+    type: str
+    description:
+     - An existing bug number to operate on.
+
+  comment:
+    required: false
+    type: str
+    description:
+     - The comment text to add.
+
+  status:
+    required: false
+    type: str
+    description:
+     - The desired status; only relevant for the transition operation.
+
+  timeout:
+    required: false
+    type: float
+    description:
+      - Set timeout, in seconds, on requests to Bugzilla API.
+    default: 10
+
+notes:
+  - "Currently this only works with REST API."
+
+author: "Chuang Cao (@cc)"
+"""
+
+EXAMPLES = """
+# Add a comment to bug:
+- name: Comment on bug
+  bugzilla:
+    uri: '{{ server }}'
+    username: '{{ user }}'
+    password: '{{ pass }}'
+    bug: '{{ bug.id }}'
+    operation: comment
+    comment: A comment added by Ansible
+
+# Retrieve metadata for an issue and use it to create an account
+- name: Get an issue
+  bugzilla:
+    uri: '{{ server }}'
+    username: '{{ user }}'
+    password: '{{ pass }}'
+    operation: fetch
+    bug: 123456
+  register: bug
+
+# Transition an issue by target status
+- name: Change bug's status to ON_QA
+  bugzilla:
+    uri: '{{ server }}'
+    username: '{{ user }}'
+    password: '{{ pass }}'
+    bug: '{{ bug.id }}'
+    operation: transition
+    status: ON_QA
+"""
+
+RETURN = '''
+bugzilla_info:
+    description: This is the is of bug
+    type: str
+    returned: always
+'''
+
+import base64
+import json
+import sys
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+
+
+# Handle request by ansible lib
+def handle_request(url, timeout, data=None, method='GET'):
+    data = json.dumps(data)
+    response, info = fetch_url(module, url, data=data, method=method, timeout=timeout,
+                               headers={'Content-Type': 'application/json'})
+    if info['status'] not in (200, 201, 204):
+        module.fail_json(msg=info['msg'])
+    body = response.read()
+    ret = {}
+    if body:
+        ret = json.loads(body)
+    return ret
+
+
+def comment(restbase, user, passwd, params):
+    data = {
+        'comment': params['comment'],
+        'Bugzilla_login': user,
+        'Bugzilla_password': passwd,
+    }
+    url = restbase + '/bug/' + params['bug'] + '/comment'
+    ret = handle_request(url, params['timeout'], data, 'POST')
+    return ret
+
+
+def fetch(restbase, user, passwd, params):
+    url = restbase + '/bug/' + params['bug'] + "?Bugzilla_login=" + user + "&Bugzilla_password=" + passwd
+    ret = handle_request(url, params['timeout'])
+    return ret
+
+
+def transition(restbase, user, passwd, params):
+    data = {
+        'status': params['status'],
+        'Bugzilla_login': user,
+        'Bugzilla_password': passwd,
+    }
+    url = restbase + '/bug/' + params['bug']
+    ret = handle_request(url, params['timeout'], data, 'PUT')
+    return ret
+
+
+# Some parameters are required depending on the operation:
```

Comment:
Use `required_if` parameter of `AnsibleModule` for that.

Reply:
Ack